### PR TITLE
graph/encoding/dot: fix line visit marking

### LINKED ISSUE #1517: graph/encoding/dot: MarshalMulti incorrectly marks lines as visited
The logic for marking visited lines does not take into account directionality of the graph and incorrectly assumes that line IDs are unique within a multigraph. While our implementations are implemented this way, the documentation merely states that "A Line returns an ID that must distinguish Lines sharing Node end points."

I have a fix, but I need to write a test for it.

Since v0.8.2 is not yet tagged, I'd like to get this fix into that release.

### PR DESCRIPTION
**Pull request by kortschak**
Please take a look.

Fixes #1517.

### CODE REVIEW

**Comment by kortschak** [comment]
*ping*

File: graph/encoding/dot/encode.go [hunk 1]
```diff
@@ -510,10 +512,18 @@ func (p *multiGraphPrinter) print(g graph.Multigraph, name string, needsIndent,
 
 			for _, l := range lines {
 				lid := l.ID()
-				if p.visited[line{inGraph: name, id: lid}] {
-					continue
+				if isDirected {
+					if p.visited[line{inGraph: name, from: nid, to: tid, id: lid}] {
+						continue
+					}
+					p.visited[line{inGraph: name, from: nid, to: tid, id: lid}] = true
```
Comment (sbinet) [inline graph/encoding/dot/encode.go:519]: perhaps construct that key once (before the `if` at line 516) and use in both places?

File: graph/encoding/dot/encode.go [hunk 2]
```diff
@@ -510,10 +512,18 @@ func (p *multiGraphPrinter) print(g graph.Multigraph, name string, needsIndent,
...
+				} else {
+					if p.visited[line{inGraph: name, from: nid, to: tid, id: lid}] {
```
Comment (sbinet) [inline graph/encoding/dot/encode.go:521]: ditto.

**Review by sbinet — DISMISSED** [review]
LGTM

(but I am no expert)

File: graph/encoding/dot/encode.go (hunk 1 above)
Comment (kortschak) [reply graph/encoding/dot/encode.go]: Done

File: graph/encoding/dot/encode.go (hunk 2 above)
Comment (kortschak) [reply graph/encoding/dot/encode.go]: Done

**Comment by kortschak** [comment]
PTAL

**Review by sbinet — APPROVED** [review]

**Pull request merged**